app/scripts/lib/crypto/recovery-keys.js

```diff
@@ -0,0 +1,176 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define(function (require, exports, module) {
+  const sjcl = require('sjcl');
+  const hkdf = require('./hkdf');
+  const hibase32 = require('hi-base32');
+
+  function importFxaCryptoDeriver() {
+    return import(/* webpackChunkName: "fxaCryptoDeriver" */ 'fxaCryptoDeriver');
+  }
+
+  const RECOVERY_KEY_VERSION = '\x01';
+  const ALPHABET_BASE32 = '0123456789ABCDEFGHJKMNPQRSTVWXYZ';
+
+  return {
+    /**
+     * Generate a random base32 string of length and
+     * prepend encryption version number.
+     *
+     * @param {Integer} length Length of string to generate
+     * @returns {string}
+     */
+    generateRecoveryKey: function (length) {
+      return importFxaCryptoDeriver().then((fxaCryptoDeriver) => {
+        const jose = fxaCryptoDeriver.jose;
+        const bytes = jose.util.randomBytes(length);
+        const code = [];
+        for (let i = 0; i < length; i++) {
+          code.push(ALPHABET_BASE32[bytes[i] % 32]);
+        }
+
+        // Prepend version and return code
+        return RECOVERY_KEY_VERSION + code.join('');
```

Comment:
With the prepended version identifier, this will return a string of length `length + 1`, which will probably be confusing for callers.  I suggested we either make the total length of the returned string `length`, or call the input parameter something else to avoid confusion.

Reply:
This returns the correct size and added test to assert it.